test/pages/AdvancedSearchPage.ts

```diff
@@ -0,0 +1,78 @@
+import {Locator, Page} from "@playwright/test"
+import { BASE_URL } from "../helper-config";
+
+export class AdvancedSearchPage {
+    page: Page;
+    hamburgerIcon: Locator;
+    hamburgerDropdownMenu: Locator;
+    advancedSearchLink: Locator;
+    advancedSearchLinkTrack: Locator;
+    advancedSearchHeading: Locator;
+    titleFieldLocator: Locator;
+    authorFieldLocator: Locator;
+    isbnFieldLocator: Locator;
+    subjectFieldLocator: Locator;
+    placeFieldLocator: Locator;
+    personFieldLocator: Locator;
+    publisherFieldLocator: Locator;
+    searchButton: Locator;
+    searchBooksHeading: Locator;
+    searchedBookName: Locator;
+    searchedBookAuthor: Locator;
+
+    constructor({page}: {page: Page}) {
+        this.page = page;
+        this.hamburgerIcon = page.getByAltText('additional options menu', {exact: true});
+        this.hamburgerDropdownMenu = page.locator('#header-bar .hamburger-dropdown-menu');
+        this.advancedSearchLink = page.locator('#header-bar .hamburger-dropdown-menu a[href="/advancedsearch"]');
+        this.advancedSearchHeading = page.getByRole('heading', {name: 'Advanced Search'});
+        this.titleFieldLocator = page.getByRole('textbox', {name: 'Title'});
+        this.authorFieldLocator = page.getByRole('textbox', {name: 'Author'});
+        this.isbnFieldLocator = page.getByRole('textbox', {name: 'ISBN'});
+        this.subjectFieldLocator = page.getByRole('textbox', {name: 'Subject'});
+        this.placeFieldLocator = page.getByRole('textbox', {name: 'Place'});
+        this.personFieldLocator = page.getByRole('textbox', {name: 'Person'});
+        this.publisherFieldLocator = page.getByRole('textbox', {name: 'Publisher'});
+        this.searchButton = page.locator('#test-body-mobile #contentBody .cta-btn--search');
+        this.searchBooksHeading = page.locator('#test-body-mobile #contentHead h1');
+        this.searchedBookName = page.locator('#searchResults .booktitle a');
+        this.searchedBookAuthor = page.locator('#searchResults .bookauthor a');
+    }
+
+    async navigateToHome() {
+        await this.page.goto(BASE_URL);
+    }
+
+    async clickOnHamburgerIcon() {
+        await this.hamburgerIcon.click();
+    }
+
+    async navigateToAdvancedSearch() {
+        await this.advancedSearchLink.click();
+    }
+
+    async advancedSearch({title, author, ISBN, subject, place, person, publisher}:{title?:string, author?:string, ISBN?:string, subject?:string, place?:string, person?:string, publisher?:string}) {
```

Comment:
what is the convention for camel case if the variable name itself is a abbreviation :man_shrugging: 
 
```suggestion
    async advancedSearch({title, author, isbn, subject, place, person, publisher}:{title?:string, author?:string, isbn?:string, subject?:string, place?:string, person?:string, publisher?:string}) {
```